File: MuxyShared/WorktreeDTO.swift
import Foundation

public struct WorktreeDTO: Identifiable, Codable, Hashable, Sendable {
    public let id: UUID
    public var name: String
    public var path: String
    public var branch: String?
    public var isPrimary: Bool
    public var createdAt: Date

    public init(
        id: UUID,
        name: String,
        path: String,
        branch: String? = nil,
        isPrimary: Bool,
        createdAt: Date
    ) {
        self.id = id
        self.name = name
        self.path = path
        self.branch = branch
        self.isPrimary = isPrimary
        self.createdAt = createdAt
    }
}

File: android/protocol/src/main/kotlin/com/muxy/protocol/dto/WorktreeDTO.kt
package com.muxy.protocol.dto

import kotlinx.serialization.Contextual
import kotlinx.serialization.EncodeDefault
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import java.time.Instant
import java.util.UUID

@OptIn(ExperimentalSerializationApi::class)
@Serializable
data class WorktreeDTO(
    val id: @Contextual UUID,
    val name: String,
    val path: String,
    val branch: String? = null,
    val isPrimary: Boolean,
    @EncodeDefault(EncodeDefault.Mode.ALWAYS)
    val canBeRemoved: Boolean = !isPrimary,
    val createdAt: @Contextual Instant,
)
